src/lib/Syntax.hs

```diff
@@ -237,6 +249,8 @@ data UExpr' = UVar UVar
 data UConDef = UConDef Name (Nest UAnnBinder)  deriving (Show, Generic)
 data UDecl = ULet LetAnn UPatAnn UExpr
            | UData UConDef [UConDef]
+           | UInterface [UType] UConDef [UAnnBinder]
+           | UInstance UType [(UVar, UExpr)]
```

Comment:
Can you please add some comments that explain what the different fields mean?

Reply:
Yes, good idea